### PR TITLE
Starts Header/Tooling for .design

### PR DESCRIPTION
**Pull request by brookescarlett**
- Standardizes our readmes further (which becomes important in the `<CodeDocs />` component)
- Defines routing for our components and the template that component pages will follow
- Adds utils folder

### CODE REVIEW

File: packages/icon/src/Icon.spec.tsx [hunk 1]
```diff
@@ -52,6 +52,7 @@ describe('packages/Icon/glyphs/', () => {
         readonly [K in string]: string | SVGNodeObject;
       };
 
+      //@ts-expect-error
```
Comment (brookescarlett) [inline packages/icon/src/Icon.spec.tsx:55]: Need to look into this more, but wanted to pass our tests :)

File: website/pages/component/[id]/index.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      readme,
+      component: id,
+    },
+  };
+};
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  const packages = fs.readdirSync('../packages');
+
+  const paths = packages.map(packageName => ({
```
Comment (hswolff) [inline website/pages/component/[id]/index.tsx:49]: nit, it's the folder name, not the package name. that's defined inside the package.json

```suggestion
  const paths = packages.map(folderName => ({
```

File: website/pages/component/[id]/index.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+  let changelogMarkdown, readme;
+
+  if (typeof id === 'string') {
+    changelogMarkdown = fs.readFileSync(
+      path.join('../packages', id, 'changelog.md'),
+    );
+
+    readme = fs
+      .readFileSync(path.join('../packages', id, 'README.md'))
```
Comment (hswolff) [inline website/pages/component/[id]/index.tsx:31]: Let's do this async instead so we don't lock the thread.

File: website/pages/component/[id]/index.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { GetStaticPaths, GetStaticProps } from 'next';
+import fs from 'fs';
+import path from 'path';
+import Layout from '../../../components/layout';
```
Comment (hswolff) [inline website/pages/component/[id]/index.tsx:5]: Do we intend this component to be re-used elsewhere?

If no, then really we should just inline it in here.

File: website/utils/index.ts [hunk 5]
```diff
@@ -0,0 +1,2 @@
+export { default as markdownToHtml } from './markdownToHtml';
```
[the remark this answers is not in the document]
Comment (brookescarlett) [reply website/utils/index.ts:1]: 👍

File: website/pages/component/[id]/index.tsx (hunk 2 above)
Comment (brookescarlett) [reply website/pages/component/[id]/index.tsx:49]: 👍

File: website/pages/component/[id]/index.tsx (hunk 4 above)
Comment (brookescarlett) [reply website/pages/component/[id]/index.tsx:5]: 👍

File: website/pages/component/[id]/index.tsx (hunk 3 above)
Comment (brookescarlett) [reply website/pages/component/[id]/index.tsx:31]: If we do this async then the prop shows up in the Component component as undefined

Comment (hswolff) [reply website/pages/component/[id]/index.tsx:63]: We have an async await below, that works fine no? May be some bug when switching to the async version.

Comment (brookescarlett) [reply website/pages/component/[id]/index.tsx:63]: That stops working when the original request isn't completed before the value is needed on line 74. I can't await the readFile function, so it's trying to convert the changelog from markdown to HTML before there's a value for the changelogMarkdown.

File: packages/modal/src/Modal.spec.tsx [hunk 6]
```diff
@@ -58,9 +58,10 @@ describe('packages/modal', () => {
       const modal = getByRole('dialog');
       fireEvent.click(modal.parentElement!);
 
-      await expect(waitForElementToBeRemoved(modal)).rejects.toEqual(
+      await expect(waitForElementToBeRemoved(modal)).rejects.toBe(
         Error('Timed out in waitForElementToBeRemoved.'),
-      );
+      ); ///
```
Comment (brookescarlett) [inline packages/modal/src/Modal.spec.tsx:63]: The error is slightly different now.

I'm seeing this: 
```
Expected value:  [Error: Timed out in waitForElementToBeRemoved.]
Received object: [Error: Timed out in waitForElementToBeRemoved.·
```

Comment (brookescarlett) [reply packages/modal/src/Modal.spec.tsx:63]: cc: @lazytype

Comment (lazytype) [reply packages/modal/src/Modal.spec.tsx:63]: my guess is node or jest dependency changed when the yarn.lock changed

File: website/pages/component/[id]/index.tsx (hunk 3 above)
Comment (hswolff) [reply website/pages/component/[id]/index.tsx]: They do provide a promise API now: https://nodejs.org/api/fs.html#fs_promise_example

Comment (brookescarlett) [reply website/pages/component/[id]/index.tsx]: used a utils wrapper but I can update to this if you think that's better

Comment (hswolff) [reply website/pages/component/[id]/index.tsx]: Def should be async please!

Comment (brookescarlett) [reply website/pages/component/[id]/index.tsx]: Yes made it async! By using util to turn fs response into a promise
`const getFileContent = util.promisify(fs.readFile);`

File: website/pages/component/[id]/index.tsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import { GetStaticPaths, GetStaticProps } from 'next';
+import fs from 'fs';
+import path from 'path';
+import util from 'util';
+import { css } from 'emotion';
+import { GridContainer, GridItem } from '../../../components/grid/Grid';
```
Comment (hswolff) [inline website/pages/component/[id]/index.tsx:7]: Maybe not for this PR, but may want to look into https://nextjs.org/docs/advanced-features/module-path-aliases to make these imports less painful.

File: website/pages/component/[id]/index.tsx [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+const topMargin = css`
+  margin-top: 36px;
+`;
+
+export default function Component({
+  component,
+  changelog,
+  readme,
+}: BaseLayoutProps) {
+  console.log(component, changelog, readme);
```
Comment (hswolff) [inline website/pages/component/[id]/index.tsx:29]: 👀

File: website/pages/component/[id]/index.tsx [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+    'utf-8',
+  );
+
+  return {
+    props,
+  };
+};
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  const packages = fs.readdirSync('../packages');
```
Comment (hswolff) [inline website/pages/component/[id]/index.tsx:77]: We should seek to do all things async as we can to make sure we don't block.

File: website/pages/component/[id]/index.tsx [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+export const getStaticProps: GetStaticProps = async ({ params }) => {
+  if (!params || typeof params.id !== 'string') {
+    return { props: {} };
+  }
+
+  const { id } = params;
+
+  const props: Partial<BaseLayoutProps> = { component: id };
+
+  const getFileContent = util.promisify(fs.readFile);
```
Comment (hswolff) [inline website/pages/component/[id]/index.tsx:58]: We should promisify outside this method so we don't re-promisify every time the method is called.

**Review by hswolff — CHANGES_REQUESTED** [review]

File: website/pages/component/[id]/index.tsx (hunk 8 above)
Comment (brookescarlett) [reply website/pages/component/[id]/index.tsx:29]: 👍

File: website/pages/component/[id]/index.tsx (hunk 7 above)
Comment (brookescarlett) [reply website/pages/component/[id]/index.tsx:7]: I already have baseURL in my tsconfig so this should just work 🤔

Comment (brookescarlett) [reply website/pages/component/[id]/index.tsx:7]: Indeed it does :)

File: website/pages/component/[id]/index.tsx (hunk 9 above)
Comment (brookescarlett) [reply website/pages/component/[id]/index.tsx:77]: 👍

File: website/pages/component/[id]/index.tsx (hunk 10 above)
Comment (brookescarlett) [reply website/pages/component/[id]/index.tsx]: 👍

**Review by hswolff — APPROVED** [review]

File: packages/icon/src/Icon.spec.tsx (hunk 1 above)
Comment (lazytype) [reply packages/icon/src/Icon.spec.tsx:55]: what changed that would even affect this file?

File: packages/confirmation-modal/src/ConfirmationModal.spec.tsx [hunk 11]
```diff
@@ -119,10 +119,10 @@ describe('packages/confirmation-modal', () => {
         requiredInputText: 'Confirm',
       });
 
-      const confirmationButton = getByText('Confirm');
+      const confirmationButton = getByText('Confirm').parentNode;
```
Comment (lazytype) [inline packages/confirmation-modal/src/ConfirmationModal.spec.tsx:122]: I propose doing `getByText('Confirm').closest('button')` to make this more robust

File: website/components/grid/Grid.tsx [hunk 12]
```diff
@@ -85,7 +85,7 @@ const baseGridItemStyles = css`
 `;
 
 const visibleGridItemStyles = css`
-  flex-grow: 1;
+  // flex-grow: 1;
```
Comment (lazytype) [inline website/components/grid/Grid.tsx:88]: rm

Comment (brookescarlett) [reply website/components/grid/Grid.tsx:88]: 👍

File: packages/confirmation-modal/src/ConfirmationModal.spec.tsx (hunk 11 above)
Comment (brookescarlett) [reply packages/confirmation-modal/src/ConfirmationModal.spec.tsx:122]: 👍

File: packages/icon/src/Icon.spec.tsx (hunk 1 above)
Comment (brookescarlett) [reply packages/icon/src/Icon.spec.tsx:55]: not sure, but the error seems to be coming from the toJson typing

**Pull request merged**